### PR TITLE
Added custom dataset schema

### LINKED ISSUE #1: Implement custom dataset Schema
In CKAN, the dataset schema lists the metadata fields present in [a dataset and resources](https://www.opendatani.gov.uk/api/action/package_show?id=notifiable-infectious-diseases-report-2016-week-17), as well as the validation rules attached to them. CKAN includes a [default schema](https://github.com/ckan/ckan/blob/master/ckan/logic/schema.py#L103) but most instances will customize it to suit their own needs.

This was traditionally done with the `IDatasetForm` interface, and it involved a fairly [tedious process](http://docs.ckan.org/en/latest/extensions/adding-custom-fields.html). Luckily now there is [ckanext-scheming](https://github.com/ckan/ckanext-scheming) available. ckanext-scheming allows to define one or more custom schemas declaratively using a JSON file like [this one](https://github.com/okfn/ckanext-unhcr/blob/master/ckanext/unhcr/schemas/dataset.json). The extension will then take care of creating any internal hooks on `IDatasetForm` and rendering the form depending on the definitions.

We might need to end up using `IDatasetForm` anyway if we need more flexibility but to start with let's try to stick with ckanext-scheming.

I already added the [schema definition file](https://github.com/okfn/ckanext-unhcr/blob/master/ckanext/unhcr/schemas/dataset.json) to this repo, which includes one of the custom fields (Data Processing Status), now we need to add the rest defined in the [Dataset Field Spec sheet](https://docs.google.com/spreadsheets/d/1Gg1lNXqZG_CBltv6hcRX6UkTmOTZdJ4MtwxmXQjY0Fk/edit#gid=90335564) and [File Field Spec sheet](https://docs.google.com/spreadsheets/d/1Gg1lNXqZG_CBltv6hcRX6UkTmOTZdJ4MtwxmXQjY0Fk/edit#gid=228295987) on the spreadsheet.

Many fields in the spec need to be clarified with the client. I added a couple of columns with comments about the status of each. We will update them as we get more details, but the ones marked `ok` and `done` are safe to add.

Tasks:

- [x] Edit the `dataset.json` file to add any existing core field and new fields needed
  * If custom validators are needed, register them with `IValidators` (example [here](https://github.com/frictionlessdata/ckanext-validation/blob/master/ckanext/validation/plugin.py#L231) and [here](https://github.com/frictionlessdata/ckanext-validation/blob/master/ckanext/validation/validators.py#L11))
  * If custom field types or template snippets are needed, create custom presets like [here](https://github.com/frictionlessdata/ckanext-validation/blob/master/ckanext/validation/examples/ckan_default_schema.json#L110) and [here](https://github.com/frictionlessdata/ckanext-validation/blob/master/ckanext/validation/presets.json)

Dataset:
- [x] Date range data collected -> #11
- [x] Date file created -> internal
- [x] Date modified -> internal
- [x] Linked datasets -> #7 
- [x] Keywords
- [x] Unit of measurement
- [x] Description
- [x] Operational purpose of data
- [x] Uploader -> internal
- [x] HXL-ated?
- [x] Raw Data Access Level
- [x] Dataset process status -> #8 
- [x] Project URL
- [x] Archive status
- [x] Identifiability -> #9 
- [x] Admin Notes -> #10 
- [x] File Attachments
- [x] Original ID
- [x] Data collection technique
- [x] License dataset

Resource:

- [x] Date range data collected -> #11 
- [x] Version
- [x] Date modified -> internal
- [x] File Type
- [x] Description
- [x] Created by -> internal
- [x] File format
- [x] HXL-ated?
- [x] File process status
- [x] Identifiability -> #9 
- [x] File Size -> internal

### PR DESCRIPTION
**Pull request by roll**
- fixes #1

### CODE REVIEW

File: ckanext/unhcr/schemas/dataset.json [hunk 1]
```diff
@@ -46,11 +40,223 @@
...
             "label": "Source"
         },
+        {
+            "field_name": "source_organization",
+            "help_text": "Which organization owns / collected the data",
+            "label": "Source organization(s)",
+            "required": true
+        },
+        {
+            "field_name": "date_range_data_collected",
```
Comment (amercader) [inline ckanext/unhcr/schemas/dataset.json]: `date_range_data_collected`,  `metadata_modified`, `metadata_created` and `author` don't need to be present as these are handled by CKAN internally

File: ckanext/unhcr/schemas/dataset.json [hunk 2]
```diff
@@ -65,9 +271,152 @@
...
+                    "value": "archived"
+                },
+                {
+                    "label": "Deleted",
+                    "value": "deleted"
+                }
+            ]
+        },
+        {
+            "field_name": "identifiability",
```
Comment (amercader) [inline ckanext/unhcr/schemas/dataset.json]: The `identifiability` at the dataset level will be automatically generated at the dataset level based on the values of all resources, so it should not be displayed in the dataset form, only in the resource form. Can we hide it from the form with `form_display: null`?

Comment (amercader) [reply ckanext/unhcr/schemas/dataset.json]: Use this [preset](https://github.com/frictionlessdata/ckanext-validation/blob/master/ckanext/validation/examples/ckan_default_schema.json#L120

File: ckanext/unhcr/schemas/dataset.json [hunk 3]
```diff
@@ -46,11 +40,223 @@
             "form_placeholder": "http://example.com/dataset.json",
             "label": "Source"
         },
+        {
+            "field_name": "source_organization",
```
Comment (amercader) [inline ckanext/unhcr/schemas/dataset.json]: This will be a list, waiting for values

File: ckanext/unhcr/schemas/dataset.json (hunk 1 above)
Comment (amercader) [reply ckanext/unhcr/schemas/dataset.json]: See #11

File: ckanext/unhcr/schemas/dataset.json [hunk 4]
```diff
@@ -46,11 +40,223 @@
...
+            "form_snippet": null
+        },
+        {
+            "field_name": "metadata_modified",
+            "help_text": "System generated field for date on which the RIDL record was last modified",
+            "label": "Date modified",
+            "form_snippet": null
+        },
+        {
+            "field_name": "linked_datasets",
```
Comment (amercader) [inline ckanext/unhcr/schemas/dataset.json]: See #7

**Comment by roll** [comment]
@amercader 
Please review. I've added list here - https://github.com/okfn/ckanext-unhcr/issues/1

I've tested dataset/resources creation - it works.

File: ckanext/unhcr/schemas/dataset.json (hunk 2 above)
Comment (roll) [reply ckanext/unhcr/schemas/dataset.json]: It says preset `hidden_in_form` is not registered. I used `form_snippet: null`.

**Comment by roll** [comment]
I will merge it now to start PRs for extracted issues. I think it should be OK because you have already reviewed it and we could update it anytime anyway.

**Pull request merged**